models/Document/Editable/Video.php

```diff
@@ -354,7 +354,7 @@ public function admin()
      */
     public function setDataFromResource(mixed $data): static
     {
-        if (!empty($data)) {
+        if (!empty($data) && is_string($data)) {
```

Comment:
what happens when i pass any value (since it accepts mixed) other than string/array? or maybe just `null`?

Reply:
@aryaantony92 please see

Reply:
@dvesh3 Yes, it will throw error in https://github.com/pimcore/pimcore/blob/87cfd9b1c1294408e50bb81081f08f3cef73395d/models/Document/Editable/Video.php#L361 if we pass null. Then,  we should recheck this case for other editables as well.

Reply:
let's focus on this one for now. Please add checks for valid types

Reply:
```suggestion
        if (is_string($data) && $data) {
```
empty() is unneeeded